modules/OFConnectionManager2/utest/main.c

```diff
@@ -546,6 +577,44 @@ tls_recvmsg(SSL *ssl, uint8_t buf[], int buflen,
     return of_object_new_from_message_preallocated(storage, buf, msglen);
 }
 
+static of_object_t *
+tls_no_recvmsg(SSL *ssl, uint8_t buf[], int buflen,
+               of_object_storage_t *storage)
+{
+    const int header_bytes = 8;
+    int ret = 0;
+    uint16_t msglen;
+    int i = 0;
+
+    do {
+        ERR_clear_error();
+        ret = SSL_read(ssl, buf, header_bytes);
+        i++;
+    } while (i < 256 && ret == -1 && 
+             SSL_get_error(ssl, ret) == SSL_ERROR_WANT_READ);
+    if ((ret == -1) &&  SSL_get_error(ssl, ret) == SSL_ERROR_WANT_READ) {
+        return NULL;
+    } else {
+        if (ret == -1) {
+            ERR_print_errors_fp(stderr);
+        }
+        INDIGO_ASSERT(ret == header_bytes, "read %d bytes, expected %d", 
+                      ret, header_bytes);
+        msglen = ntohs(*((uint16_t*)(buf+2)));
+        INDIGO_ASSERT(msglen <= buflen, "message len %d exceeds buffer len %d",
+                      msglen, buflen);
+        if (msglen - header_bytes > 0) {
+            ERR_clear_error();
+            ret = SSL_read(ssl, buf+header_bytes, msglen-header_bytes);
+            INDIGO_ASSERT(ret == msglen-header_bytes, 
+                          "body read %d bytes, expected %d",
+                          ret, msglen-header_bytes);
+        }
+
+        return of_object_new_from_message_preallocated(storage, buf, msglen);
+    }
+}
```

Comment:
the same comment that i wrote for tcp_recvmsg and tcp_no_recvmsg also applies here